feat(kno-5292): docs for slack-kit (#408)
knocklabs/docs: content/integrations/chat/slack/slack-kit-trigger-workflow.mdx

---
title: Trigger workflow
description: Triggering workflows with SlackKit
tags: ["slack", "chat"]
section: Integrations > Slack > SlackKit > Trigger workflow
layout: integrations
---

> There is very little to adjust about the way you handle triggering notifications when opting to use Knock's SlackKit for your Slack integration. The main change is that every time you trigger a workflow for the Slack workspace connected using SlackKit, you'll be passing the tenant you used in the SlackAuthButton as the `tenant` in the data payload. That's because during the authentication process, your tenant stored the Slack access token necessary to pull data as well as post notifications to Slack channels. So when you use it in your trigger, you can post to the Slack channels stored as channel data on your recipient without that channel data needing their own access token.

Reviewer: ```suggestion
When you trigger a workflow for the Slack workspace connected using SlackKit, you'll pass the tenant you used in the SlackAuthButton as the `tenant` in your workflow trigger. This is because during the OAuth process, your tenant stored the Slack access token necessary to post notifications to Slack. When you use that tenant in your trigger, we use the access token stored on that tenant to post to the Slack channels stored on your recipients.
```

> Let's take an example. You want to trigger a notification when someone posts a comment to your `VideoPage` about a specific video.

Reviewer 2: ```suggestion
Let's take a look at an example: you want to trigger a notification when someone posts a comment to your `VideoPage` about a specific video.
```

    "channelData": {
        "channel_id": "<Knock Slack channel ID>",
        "data": {
            "connections": [],
            "token": {
                "access_token": "xoxo-123456789"
            }
        }
    }
}

// Recipient object
{
    "id": "dinosaurs-loose",
    "collection": "videos",
    "channelData": {
        "channel_id": "<Knock Slack channel ID>",
        "data": {
            "connections": [
                {
                    "channel_id": "CMYK12345", // The id of the Slack channel in your user's slack workspace
                    "access_token": null, // Will inherit the access token from the tenant
                    "user_id": null, // Not a direct message
                    "incoming_webhook": null // Uses access token instead of incoming webhook to post messages
                },
                // Can post to multiple Slack channels in the same workspace at once
                {
                    "channel_id": "CXYZ54321",
                    "access_token": null,
                    "user_id": null,
                    "incoming_webhook": null
                }
            ],
            "token": null
        }
    }
}
```

> Now that you know how to use the full Knock-managed flow, we can look at further customization by opting out of the components and using the pieces SlackKit offers under the hood to build your own components and flows on top of.

Reviewer 2: ```suggestion
Now that you know how to use the full Knock-managed flow, we can look at further customization by opting out of the components and using the hooks and APIs SlackKit offers under the hood to build your own components and flows on top of.
```
